packages/core/src/api/webhook.ts

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable jsdoc/check-param-names */
+import { makeURLSearchParams, type REST, type RawFile } from '@discordjs/rest';
+import {
+	Routes,
+	type APIMessage,
+	type APIWebhook,
+	type RESTPatchAPIWebhookJSONBody,
+	type RESTPatchAPIWebhookWithTokenMessageJSONBody,
+	type RESTPostAPIChannelWebhookJSONBody,
+	type RESTPostAPIWebhookWithTokenGitHubQuery,
+	type RESTPostAPIWebhookWithTokenJSONBody,
+	type RESTPostAPIWebhookWithTokenQuery,
+	type RESTPostAPIWebhookWithTokenSlackQuery,
+} from 'discord-api-types/v10';
+
+export class WebhooksAPI {
+	public constructor(private readonly rest: REST) {}
+
+	/**
+	 * Fetches a webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 */
+	public async get(id: string, token?: string) {
+		return this.rest.get(Routes.webhook(id, token));
+	}
+
+	/**
+	 * Creates a new webhook
+	 *
+	 * @param channelId - The id of the channel to create the webhook in
+	 * @param options - The options to use when creating the webhook
+	 * @param reason - The reason for creating the webhook
+	 */
+	public async create(channelId: string, options: RESTPostAPIChannelWebhookJSONBody, reason?: string) {
+		return (await this.rest.post(Routes.channelWebhooks(channelId), {
+			reason,
+			body: options,
+		})) as APIWebhook;
+	}
+
+	/**
+	 * Edits a webhook
+	 *
+	 * @param id - The id of the webhook to edit
+	 * @param webhook - The new webhook data
+	 * @param token - The token of the webhook
+	 * @param reason - The reason for editing the webhook
+	 */
+	public async edit(id: string, webhook: RESTPatchAPIWebhookJSONBody, token?: string, reason?: string) {
+		return (await this.rest.patch(Routes.webhook(id, token), {
+			reason,
+			body: webhook,
+		})) as APIWebhook;
+	}
+
+	/**
+	 * Deletes a webhook
+	 *
+	 * @param id - The id of the webhook to delete
+	 * @param token - The token of the webhook
+	 * @param reason - The reason for deleting the webhook
+	 */
+	public async delete(id: string, token?: string, reason?: string) {
+		return this.rest.delete(Routes.webhook(id, token), { reason });
+	}
+
+	/**
+	 * Executes a webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 * @param options - The options to use when executing the webhook
+	 */
+	public async execute(
+		id: string,
+		token: string,
+		options: RESTPostAPIWebhookWithTokenJSONBody & RESTPostAPIWebhookWithTokenQuery & { files?: RawFile[] } = {},
+	) {
+		const { wait, thread_id, files, ...body } = options;
```

Comment:
Should this actually be optional?
```suggestion
		{ wait, thread_id, files, ...body }: RESTPostAPIWebhookWithTokenJSONBody & RESTPostAPIWebhookWithTokenQuery & { files?: RawFile[] },
	) {
```